db/types/operations/cast.py

```diff
@@ -558,10 +622,26 @@ def _get_date_type_body_map():
     # Note that default postgres conversion for dates depends on the
     # `DateStyle` option set on the server, which can be one of DMY, MDY,
     # or YMD. Defaults to MDY.
-    default_behavior_source_types = frozenset([DATE]) | TEXT_TYPES
-    return _get_default_type_body_map(
-        default_behavior_source_types, DATE,
+    source_text_types = TEXT_TYPES
+    default_behavior_source_types = frozenset([DATE])
```

Comment:
I think `TIMESTAMP` types should also be valid sources for dates (in case there's a timestamp value at midnight so the cast would be lossless).